Update for Migration to UVic's Arbutus Cloud / Kubernetes Cluster

aaronskiba opened the pull request:
Changes proposed in this PR:

### `Dockerfile.production`
- Implemented multi-stage build for improved efficiency and reduced image size
  - `wkhtmltopdf` stage pulls a `surnet/alpine-wkhtmltopdf` Docker image
  - `builder` stage steps include performing yarn install and precompilation of assets as non-root user
  - `app` and `assets` stages pertain to production runtime of Rails app and serving of static assets via nginx respectively
- Switched to an Alpine base image for a smaller and more secure container
- Added a healthcheck using the `/api/v1/heartbeat` path
- Configured BUNDLE_WITHOUT to exclude non-production groups during `bundle install`, keeping the image lean
- NOTE: Production environment setup requires the following environmental variables:
  ```
  BUNDLE_PATH: 'vendor/bundle'
  WICKED_PDF_PATH: '/usr/bin/wkhtmltopdf'
  BUNDLE_WITHOUT: 'development:test:mysql:aws:sandbox:ci:wkhtmltopdf'
  RAILS_ENV: production
  NODE_ENV: production
  LC_ALL: en_CA.UTF-8
  ```
### `.github/workflows/docker-push-image.yml` & `.github/workflows/docker-push-uat-image.yml`
  - Modified workflows to each publish two Docker images:
    - `-app` image corresponding to the Rails app
    - `-assets` image corresponding to the nginx asset host
  - Added `docker/setup-buildx-action@v2` for advanced Docker build features like multi-platform builds and layer caching
  - Configured build cache to push to / fetch from Docker Hub to speed up image builds

### All `.github/workflows/` Files
  - Bumped node-version: to '20.15.1'
    - This aligns with `ARG NODE_VERSION=20.15.1-r0` now specified in `Dockerfile.production` (This version was simply the default one being installed before specifying it within `Dockerfile.production`.).

### `nginx.conf` & `nginx.conf.template`
  - Added configuration files for nginx to serve the Rails assets
  
###  `Gemfile` & `Gemfile.lock`
  - Added `, group: :wkhtmltopdf` to `gem 'wkhtmltopdf-binary'`. This gem is not compatible with Alpine Linux. To address that, 'wkhtmltopdf' has been added to `BUNDLE_WITHOUT`.
  - Add `x86_64-linux-musl` platform to Gemfile.lock 1ad1de4ff4768f499a1b4dcc03891945d8d887a2
    - `bundle lock --add-platform x86_64-linux-musl` adds platform for compatibility with alpine precompiled binaries

### `config/environments/` files (`production`, `staging`, and `sandbox`)
- Redirect logs to $stdout
  - This change aligns with best practices for containerised applications.
  - Removed `gem 'syslog-logger'` in the process

### `docker-compose.yml`
- Updated and added web service to help simulate a production environment locally.

### `.dockerignore`
  - Added entries for non-production-related files to reduce image size

### `deployments/staging/*-secret-sealed.yaml` files
  - Encrypted staging secrets via `kubeseal` (Secrets can only be decrypted by the Kubernetes cluster)

aaronskiba committed:
Redirect prod/staging/sandbox logs to $stdout

This change aligns with best practices for containerised applications.

aaronskiba committed:
Rework Docker image pipeline

Breakdown of changes by file:
- `.dockerignore`
  - Added entries for non-production-related files to reduce image size

- `.github/workflows/docker-push-image.yml` and `.github/workflows/docker-push-uat-image.yml`
  - Added `docker/setup-buildx-action@v2` for advanced Docker build features like multi-platform builds and layer caching
  - Enabled caching of gem dependencies to speed up builds
  - Configured build cache to push to / fetch from Docker Hub to speed up image builds
  - NOTE: To adhere to DRY principles, these two files should be combined into one

- `Dockerfile.production`
  - Implemented multi-stage build for improved efficiency and reduced image size
  - Switched to an Alpine base image for a smaller and more secure container
  - Added a healthcheck using the `/api/v1/heartbeat` path
  - Configured BUNDLE_WITHOUT to exclude non-production groups during `bundle install`, keeping the image lean
  - NOTE: Production environment setup requires the following environmental variables:
    - BUNDLE_PATH: 'vendor/bundle'
    - WICKED_PDF_PATH: '/usr/bin/wkhtmltopdf'
    - BUNDLE_WITHOUT: 'development:test:mysql:aws:sandbox:ci:wkhtmltopdf'
    - RAILS_ENV: production
    - NODE_ENV: production

- `Gemfile`
  - Added `, group: :wkhtmltopdf` to `gem 'wkhtmltopdf-binary'`. This gem is not compatible with Alpine Linux. To address that, 'wkhtmltopdf' has been added to `BUNDLE_WITHOUT`.

aaronskiba committed:
add `x86_64-linux-musl` platform to Gemfile.lock

`bundle lock --add-platform x86_64-linux-musl` adds platform for compatibility with alpine precompiled binaries

aaronskiba committed:
Copy docker-compose-development to docker-compose

aaronskiba committed:
Add web service to `docker-compose.yml`

Configured to simulate a production environment for local testing.

aaronskiba committed:
Copy required `Node` libraries to prod image

Prior to this commit, errors were encountered when executing `node -v`.  This commit copies the required `Node.js` libraries from the builder stage, thereby ensuring they are available within the production image.

aaronskiba committed:
Fix bold font / update wkhtmltopdf deps handling

Replaced the `COPY --from=wkhtmltopdf /usr/lib/ /usr/lib/` approach with direct installation of necessary dependencies in the Dockerfile's production stage. The specified dependencies are copied directly from https://github.com/Surnet/docker-wkhtmltopdf/pkgs/container/alpine-wkhtmltopdf#other-images .

This change resolves a text styling issue where text was failing to render in bold.

aaronskiba committed:
Install Node.js in prod stage of Dockerfile

- Directly install Node.js within the Dockerfile production stage.
- Because Node.js is also installed within the builder stage, we previously tried copying the binary and its libraries from that stage into the production stage. However, directly installing it again within the production stage simplifies the setup and ensures all dependencies are installed correctly. (The commits 809c832fba73ff776b2f0918714930636fa17b07 and cff19b26f477a9f8f514021d6f199855efb3ed5a fix issues encountered when applying this COPY approach to wkhtmltopdf and Node.js).

aaronskiba committed:
Update to latest `surnet/alpine-wkhtmltopdf`

aaronskiba committed:
Match .github/workflows node-version to Dockerfile

aaronskiba committed:
Refactor Dockerfile.production deps and versioning

aaronskiba committed:
Remove `gettext` runtime ENV vars from Dockerfile

aaronskiba committed:
Add node + postgresql dep versions && `apk update`

- Added `apk update` to address warnings related to outdated package lists.
- The chosen nodejs and postgresql versions are the current default ones, which will help in maintaining stability across builds.

aaronskiba committed:
Update docker-compose env vars

aaronskiba committed:
Add deps & dir to enable `dragonfly` image upload

aaronskiba committed:
Set separate web & api stages within Dockerfile

- Split the Dockerfile's `production` stage into separate `api` and `web` stages
  - `web` handles static assets via Nginx and proxies dynamic requests to `api` (Rails app)
- Created `nginx.conf` to configure Nginx to serve static assets and proxy dynamic requests to the Rails API.
- Updated GitHub Actions workflows to build and push both the Web (Nginx) and API (Rails) Docker images to DockerHub.

dependabot[bot] committed:
Bump rack from 2.2.10 to 2.2.11

Bumps [rack](https://github.com/rack/rack) from 2.2.10 to 2.2.11.
- [Release notes](https://github.com/rack/rack/releases)
- [Changelog](https://github.com/rack/rack/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rack/rack/compare/v2.2.10...v2.2.11)

---
updated-dependencies:
- dependency-name: rack
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

aaronskiba committed:
Temp: Disable Docker healthcheck & _ nginx.conf

- Commented out Docker healthcheck to avoid unnecessary retries.
- Set Nginx server_name to wildcard (_) to allow all incoming requests.
- Commented out build args for simplicity in initial push.
This commit will be reverted once the relevant build args are finalised and handled properly.

aaronskiba committed:
Use runtime vars for healtcheck & nginx config

- Re-enabled Docker Healthcheck, now using dynamic runtime `BASE_URL` environmental var
- Replaced wildcard(_) in nginx.conf with dynamic runtime  `SERVER_NAME` environmental var
- Removed unnecessary build-args from Docker-related .github/workflows files.

These changes revert the temporary changes made in commit d19ab7eaade4c7bf987906eb8a429064716d98ad, ensuring a more secure and flexible setup.

aaronskiba committed:
Simplify caching in Docker image workflows

Removed the "Cache Gem dependencies" step from both Docker image workflows.
- Bundler dependencies (`vendor/bundle`) were previously cached using GitHub Actions, but since Docker’s `cache-to` and `cache-from` already handle these layers, the step appears to be redundant.

aaronskiba committed:
Update Docker tagging for staging vs prod images

Removed `latest` and `production` tag generation for both the "-api" and "-web" tags
Implemented the following tagging logic:
- if release_type == prerelease && Git tag matches `.+portage-(\d+\.\d+\.\d+-alpha-\d+)$` pattern, then generate the following tags:
  - `x.y.z-alpha-n-api` and `x.y.z-alpha-n-web`
- else if release_type == latest && Git tag matches `.+portage-(\d+\.\d+\.\d+)$` pattern, then generate the following tags:
  - `x.y.z-api` and `x.y.z-web`

aaronskiba committed:
Specify nginx alpine version in Dockerfile.production

aaronskiba committed:
Set nginx.conf server_name back to wildcard

aaronskiba committed:
Enhance security & K8s compatibility in Dockerfile

Thanks to Shaun Bathgate for these code change recommendations:
- Set numeric UID (1000) and GID (3000) for dmpuser and dmpgroup to avoid collisions in Kubernetes.
- Ensure all necessary directories have the correct ownership and permissions.
- Adjust permissions to avoid the need for root or escalation.
- Update nginx configuration to run as non-root user (dmpuser).
- Ensure nginx.pid is writable by dmpuser.
- Use non-root user in builder stage and precompile assets as non-root user.

aaronskiba committed:
Set API_URL dynamically in `nginx.conf`

- Uses `envsubst` at runtime to replace `${API_URL}` in `nginx.conf`, allowing different values for `proxy_pass` in Kubernetes vs Docker environments.
- Ensures `dmpuser` has the necessary permissions to modify `nginx.conf` by setting correct ownership and write access.
- After substitution, updates permissions to 644, making `nginx.conf` read-only to prevent further modifications.
- Avoids permission issues by writing directly to `/etc/nginx/nginx.conf` using `tee` instead of `mv`.
- Keeps Nginx running in the foreground with `exec nginx -g 'daemon off;'`.

aaronskiba committed:
Follow best practices for `nginx.conf`

Thanks to Shaun Bathgate for these code change recommendations:
- NOTE: This will have to be tested within the Kubernetes cluster.
- NOTE: `proxy_set_header X-Forwarded-Proto http_x_forwarded_proto;` may need to be `proxy_set_header X-Forwarded-Proto $scheme;`
- NOTE: `user dmpuser;` is absent from the file because including it results in the following warning:
  - `[warn] 1#1: the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:2`

aaronskiba committed:
Update CSF: Allow Google Fonts, GTM, GA, and ORCID

- Updated Content-Security-Policy (CSP) to allow Google Tag Manager (GTM) and Google Analytics (GA).
- Added https://www.googletagmanager.com and https://www.google-analytics.com to script-src.
- Allowed Google Fonts by adding https://fonts.googleapis.com to style-src and https://fonts.gstatic.com to font-src.
- Ensured connect-src includes https://www.google-analytics.com for analytics data tracking.
- Added img-src 'self' https://orcid.org; to permit ORCID images

aaronskiba committed:
Update CSP to Allow ORCID Image Redirects

- Added https://info.orcid.org to img-src to account for ORCID’s domain redirection
- Prior to this change, the ORCID logo would not render on the `/users/edit` page

aaronskiba committed:
Update CSP for Full reCAPTCHA Support

aaronskiba committed:
Use ENTRYPOINT for Nginx runtime substitution

aaronskiba committed:
Merge pull request #1018 from portagenetwork/dependabot/bundler/rack-2.2.11

Bump rack from 2.2.10 to 2.2.11

aaronskiba committed:
Merge branch 'integration' into docker-pipeline

aaronskiba committed:
Add staging/ SealedSecrets & update .gitignore

- Added SealedSecrets for the staging environment under deployments/staging/
- Update `.gitignore` and `.dockerignore` to exclude sensitive files used to to generate SealedSecrets

aaronskiba committed:
Refactor nginx.conf for envsubst automation

Huge thank you to Shaun Bathgate for the following improvements:
- Refactored Nginx configuration:
 - Separated base config from the server config to support envsubst
- Moved `nginx.conf.template` to `/etc/nginx/templates/`
  - The official Nginx image automatically processes templates in this folder using envsubst.
- Updated nginx.conf
  - Set pid /tmp/nginx.pid; for better compatibility in containerized environments.
  - Included include /etc/nginx/conf.d/*.conf; to allow dynamic population from templates.

aaronskiba committed:
Fix Rails CSRF issue/Preserve original host header

- This ensures Rails correctly receives the original Host header, including the port if present.
- Resolves ActionController::InvalidAuthenticityToken errors caused by mismatched Origin and request.base_url (error below):
api-1  | HTTP Origin header (http://localhost:8080) didn't match request.base_url (http://localhost)
api-1  |
api-1  | ActionController::InvalidAuthenticityToken (ActionController::InvalidAuthenticityToken):
- Intended to prevent future issues when running behind a proxy like Traefik by preserving the client’s original request host.

aaronskiba committed:
Allow Google Maps embeds in CSP

These changes enable the Google Map rendering on `/contact-us` path

aaronskiba committed:
Refactor Nginx CSP / improve readability

aaronskiba committed:
Fix lingering CSP issue with Google Tag Manager

dependabot[bot] committed:
Bump net-imap from 0.4.12 to 0.4.19

Bumps [net-imap](https://github.com/ruby/net-imap) from 0.4.12 to 0.4.19.
- [Release notes](https://github.com/ruby/net-imap/releases)
- [Commits](https://github.com/ruby/net-imap/compare/v0.4.12...v0.4.19)

---
updated-dependencies:
- dependency-name: net-imap
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

aaronskiba committed:
Update integrations-secret-sealed.yaml

aaronskiba committed:
Remove unused `config.direct_link_protocol`

- This code was originally added as part of the following commit: https://github.com/portagenetwork/roadmap/commit/4d51c81f89d457c9ace8ac9cf4abef54091002e4
- However, with the exception of this definition in `config/application.rb`, all other references to it have been removed from the codebase.
- Our new Traefik + nginx setup will be configured to perform the proper protocol handling when passing to the Rails app.

aaronskiba committed:
Update CHANGELOG.md

aaronskiba committed:
Add missing `rack` entry to CHANGELOG

- Adds CHANGELOG entry in reference to commit 937278cbb2fadedc7c2bdc8d88f5983b8b4fb601

aaronskiba committed:
Merge pull request #1008 from portagenetwork/dependabot/bundler/net-imap-0.4.19

Bump net-imap from 0.4.12 to 0.4.19

dependabot[bot] committed:
Bump rack from 2.2.11 to 2.2.12

Bumps [rack](https://github.com/rack/rack) from 2.2.11 to 2.2.12.
- [Release notes](https://github.com/rack/rack/releases)
- [Changelog](https://github.com/rack/rack/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rack/rack/compare/v2.2.11...v2.2.12)

---
updated-dependencies:
- dependency-name: rack
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

aaronskiba committed:
Update CHANGELOG.md

aaronskiba committed:
Merge pull request #1023 from portagenetwork/dependabot/bundler/rack-2.2.12

Bump rack from 2.2.11 to 2.2.12

dependabot[bot] committed:
Bump serialize-javascript from 6.0.1 to 6.0.2

Bumps [serialize-javascript](https://github.com/yahoo/serialize-javascript) from 6.0.1 to 6.0.2.
- [Release notes](https://github.com/yahoo/serialize-javascript/releases)
- [Commits](https://github.com/yahoo/serialize-javascript/compare/v6.0.1...v6.0.2)

---
updated-dependencies:
- dependency-name: serialize-javascript
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

aaronskiba committed:
Update CHANGELOG.md

aaronskiba committed:
Merge pull request #1015 from portagenetwork/dependabot/npm_and_yarn/serialize-javascript-6.0.2

Bump serialize-javascript from 6.0.1 to 6.0.2

dependabot[bot] committed:
Bump nokogiri from 1.16.8 to 1.18.3

Bumps [nokogiri](https://github.com/sparklemotion/nokogiri) from 1.16.8 to 1.18.3.
- [Release notes](https://github.com/sparklemotion/nokogiri/releases)
- [Changelog](https://github.com/sparklemotion/nokogiri/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sparklemotion/nokogiri/compare/v1.16.8...v1.18.3)

---
updated-dependencies:
- dependency-name: nokogiri
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

aaronskiba committed:
Update CHANGELOG.md

aaronskiba committed:
Merge pull request #1024 from portagenetwork/dependabot/bundler/nokogiri-1.18.3

Bump nokogiri from 1.16.8 to 1.18.3

dependabot[bot] committed:
Bump uri from 0.13.0 to 0.13.2

Bumps [uri](https://github.com/ruby/uri) from 0.13.0 to 0.13.2.
- [Release notes](https://github.com/ruby/uri/releases)
- [Commits](https://github.com/ruby/uri/compare/v0.13.0...v0.13.2)

---
updated-dependencies:
- dependency-name: uri
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

aaronskiba committed:
Update CHANGELOG.md

aaronskiba committed:
Merge pull request #1025 from portagenetwork/dependabot/bundler/uri-0.13.2

Bump uri from 0.13.0 to 0.13.2

aaronskiba committed:
Merge branch 'deployment-portage' into docker-pipeline

aaronskiba committed:
Remove api proxying from `nginx.conf.template `

Removed proxy configuration, delegating traffic routing to Traefik.

aaronskiba committed:
Enable setting of `asset_host` via ENV

- Note, whatever value is assigned here will likely have to be assigned within `DMPROADMAP_HOST` as well

aaronskiba committed:
Exclude "healthz" path from host auth

- Updated Rails host auth config to exclude the "/healthz" endpoint from DNS rebinding checks. This ensures Kubernetes readiness checks can pass successfully when pods are coming online.

aaronskiba committed:
nginx.conf.template / api proxy removal changes

Changes provided by Shaun Bathgate along with commit 76b373a2785b79bb8ca1b252e6ff29afec3ae5a6

aaronskiba committed:
Fix: Exclude `api/v1/heartbeat` (not `healthz`)

`api/v1/heartbeat` is what we want for the healthcheck request path

aaronskiba committed:
Add `RAILS_RELATIVE_URL_ROOT` to nginx assets path

aaronskiba committed:
Remove  location /healthz from nginx.conf.template

aaronskiba committed:
Revert "Remove  location /healthz from nginx.conf.template"

This reverts commit 122f9275051c6e2b8d712c1f8cec86c4eb38a530.

aaronskiba committed:
Remove `try_files $uri =404;` from nginx template

aaronskiba committed:
Fix serving of assets via nginx

Changes and notes provided by Shaun Bathgate:
•	^/dmp-pgd-staging/assets/(.*)$ captures everything after /assets/ into $1.
•	alias /usr/share/nginx/html/assets/; means Nginx maps $1 → /usr/share/nginx/html/assets/<$1>.
•	try_files $1 =404; ensures that if <filename> doesn’t exist, you get a 404 – and if it does exist, you get the file directly.

aaronskiba committed:
`RAILS_RELATIVE_URL_ROOT` env in static asset URLs

Might be best to add these files to assets pipeline in the future.

aaronskiba committed:
Bake-in relative root url when precompiling assets

By using the RAILS_RELATIVE_URL_ROOT argument at build time, the asset pipeline precompiles with an embedded base URL. This approach is to ensure that all asset references are aligned with the required subdirectory.

dependabot[bot] committed:
Bump rack from 2.2.12 to 2.2.13

Bumps [rack](https://github.com/rack/rack) from 2.2.12 to 2.2.13.
- [Release notes](https://github.com/rack/rack/releases)
- [Changelog](https://github.com/rack/rack/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rack/rack/compare/v2.2.12...v2.2.13)

---
updated-dependencies:
- dependency-name: rack
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

aaronskiba committed:
Update CHANGELOG.md

aaronskiba committed:
Merge pull request #1026 from portagenetwork/dependabot/bundler/rack-2.2.13

Bump rack from 2.2.12 to 2.2.13

aaronskiba committed:
Merge branch 'deployment-portage' into docker-pipeline

aaronskiba committed:
Fix duplicate URL prefix in ORCID OAuth callback

Explicitly set `redirect_uri` in OmniAuth ORCID configuration to prevent duplication
of the relative URL root (`RAILS_RELATIVE_URL_ROOT`) in OAuth callback URLs.

Previously, relying on OmniAuth's automatic URL construction caused repeated path segments
(e.g., `/RAILS_RELATIVE_URL_ROOT/RAILS_RELATIVE_URL_ROOT/...`), leading to OAuth failures
due to mismatched registered callback URLs.

aaronskiba committed:
Fix CILogon sign-in 404 / handle referer path

This change ensures that the referrer path includes the correct relative root URL when signing in via CILogon. Previously, signing in through CILogon was causing a 404 error due to a missing relative root in the referrer path.

aaronskiba committed:
Fix final 404s/temp hardcode relative URL root

This temporary patch will have to be revisited/improved in the near future.

aaronskiba committed:
Rename Dockerfile stages

Renamed api to app
Renamed web to assets

aaronskiba committed:
Update docker-push workflows naming

Renamed api to app
Renamed web to assets

aaronskiba committed:
Use semver for versioning uat images

sbathgate committed:
feat(chart): initialize dmp-roadmap helm chart with core configuration and templates

sbathgate committed:
add cephfs monitors

aaronskiba committed:
Move Rails start CMD to Dockerfile

This change simplifies our setup by allowing the same command to be removed from individual orchestration tools (docker-compose kubernetes deployment, etc.).

aaronskiba committed:
Revert "Fix final 404s/temp hardcode relative URL root"

This reverts commit a2b174080a373724c908b8e0e91429bf966f3bb9.

aaronskiba committed:
Revert "Fix CILogon sign-in 404 / handle referer path"

This reverts commit 6177e411e52e72e2f7086984c61cb33cedf0154e.

aaronskiba committed:
Revert "Fix duplicate URL prefix in ORCID OAuth callback"

This reverts commit 95fc9e271d920c25d4dba37dc3c52c0ef1d6a2dc.

aaronskiba committed:
Revert "Bake-in relative root url when precompiling assets"

This reverts commit 51e31292128a317b93dee009a4a62a3198ada560.

aaronskiba committed:
Revert "`RAILS_RELATIVE_URL_ROOT` env in static asset URLs"

This reverts commit 97f3a5945bb5278139f4f51fe09abc312220f2f6.

aaronskiba committed:
Remove relative root from nginx.conf.template

aaronskiba committed:
Remove static `uat` tags from docker uat workflow

We are already generating the `-latest` tags for the app and assets images (which we may eventually delete as well). This commit removes the `uat` tags which seem redundant.

aaronskiba committed:
Merge branch 'integration' into docker-pipeline

aaronskiba committed:
Merge pull request #1028 from sbathgate/sb/kubernetes-configuration

feat(chart): initialize dmp-roadmap helm chart with core configuratio…

aaronskiba committed:
Add `deployments/prod/sealed-secrets/` files

sbathgate committed:
refactor(config): streamline default URL configuration

lagoan reviewed (dismissed):
This is really good job. It is great that you have some todo's here to come back to.

Lets pass this PR as is and come back to review the code together at a later time.

![LGTM](https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExZW0wM3VrcDM4d2U1ajdsZnBoNHV6Z2Jmdzd3N2R2N29rOGZ6aGtoeCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/tIeCLkB8geYtW/giphy.gif)

aaronskiba committed:
Combine docker workflows  &  update tag patterns

- These changes include a refactor that enable the removal of `.github/workflows/docker-push-uat-image.yml` by implementing its logic into `.github/workflows/docker-push-image.yml`.
- Additionally, the `-uat-` tagging stucture has been replaced with `-dev.` and `-alpha-` with `-rc.`
- Finally, the changes from b7497898a77902049da9878d14e3bbf0c33d3464 have been undone, so `-dev.` tags now follow the `dev.n` structure rather than `dev.x.y.z`.

sbathgate committed:
fix(deployments): update image tags for staging to latest development version

sbathgate committed:
fix(submodules): disable recursive fetching for data cleanup rules

sbathgate committed:
feat(ingress): implement dynamic hostname configuration for ingress rules

sbathgate committed:
fix(ingress): remove redundant host range loop from ingress rules

sbathgate committed:
feat(ingress): add TLS secret name configuration for ingress resources

sbathgate committed:
fix(ingress): ensure correct YAML indentation for hostname in ingress configuration

sbathgate committed:
feat(ingress): update cluster issuer for TLS configurations in production and staging environments

sbathgate committed:
fix(ingress): change cluster issuer to issuer for cert-manager annotations

sbathgate committed:
update helm-docs

sbathgate committed:
feat(cronjobs): update schedules for research data output and translation sync

aaronskiba committed:
Merge pull request #1038 from sbathgate/sb/update-dmp-chartx

Refactor: Implement Host-Based Routing and Remove Subpath URL Configuration

aaronskiba committed:
Update CHANGELOG.md

lagoan reviewed (approved):
LGTM!
